docs/specs/architecture.md

```
@@ -206,44 +264,85 @@ type ErrorCode = 'VALIDATION_ERROR' | 'AUTH_ERROR' | 'API_ERROR' | 'NETWORK_ERRO
+- Background：`chrome.tabs.onRemoved` で flag の `startedAt` を確認。TTL 超過なら cleanup （幽霊フラグ防止）
+- この二重チェックにより、どちらか一方が失敗した際もリソースリークを防止
 
 純粋関数 (`extractPageNumber`, `buildPageUrl`, `mergeBooks` 等) は `apps/extension/src/content/shared/scrape-session.ts` に集約してテスト容易性を確保。
 
 ### エラーハンドリング
 
-```typescript
-// 各エラーケースは ErrorCode で分類
+**内部メッセージエラー (Content Script ↔ Background):**
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)